app/controllers/api/v1/restaurants_controller.rb

```diff
@@ -3,16 +3,31 @@
 
 class Api::V1::RestaurantsController < ApplicationController
   def index
-    first_result = Geocoder.search(params[:location])[0]
-    lat = first_result.data["lat"].to_f
-    lon = first_result.data["lon"].to_f
+    if Search.where(search: params[:location]).empty?
+      search = Search.new(search: params[:location])
+      
+      if !search.save
+        return render json: 'Given empty location'
+      end
+
+      first_result = Geocoder.search(params[:location])[0]
+      lat = first_result.data["lat"].to_f
+      lon = first_result.data["lon"].to_f
+
+      longitude_diff = 0.01  # about 0.5 miles
+      latitude_diff = 0.01   # about 0.5 miles
 
-    longitude_diff = 0.01
-    latitude_diff = 0.01
+      open_street_map_restaurants = get_restaurants(lat, lon, latitude_diff, longitude_diff)
 
-    open_street_map_restaurants = get_restaurants(lat, lon, latitude_diff, longitude_diff)
+      render json: filter_restaurants(open_street_map_restaurants, search)
+    else
+      render json: Search.find_by(search: params[:location]).restaurants
+    end
+  end
```

Comment:
this is a very interesting pattern, specifically saving a Search. Ill def allow this mind you, but note that the restaurant data would get stale over time